src/main/java/org/victorrobotics/frc/dtlib/controller/DTXboxController.java

```diff
@@ -0,0 +1,391 @@
+    public final DTTrigger bButton;
+    public final DTTrigger xButton;
+    public final DTTrigger yButton;
+    public final DTTrigger leftBumper;
+    public final DTTrigger rightBumper;
+    public final DTTrigger leftMenuButton;
+    public final DTTrigger rightMenuButton;
+    public final DTTrigger leftStickButton;
+    public final DTTrigger rightStickButton;
+
+    public final DTTrigger dpadNone;
+    public final DTTrigger dpadUp;
+    public final DTTrigger dpadUpRight;
+    public final DTTrigger dpadRight;
+    public final DTTrigger dpadDownRight;
+    public final DTTrigger dpadDown;
+    public final DTTrigger dpadDownLeft;
+    public final DTTrigger dpadLeft;
+    public final DTTrigger dpadUpLeft;
+
+    public final DTAxis leftStickX;
+    public final DTAxis leftStickY;
+    public final DTAxis rightStickX;
+    public final DTAxis rightStickY;
+    public final DTAxis leftTrigger;
+    public final DTAxis rightTrigger;
+
+    private double leftRumblePower;
+    private double rightRumblePower;
+    private long   leftRumbleTimeout;
+    private long   rightRumbleTimeout;
+    private double axisDeadBand;
+
+    public DTXboxController(int port) {
+        controller = new XboxController(port);
+        axisDeadBand = AXIS_DEADBAND;
+        RUMBLE_TIMER.scheduleAtFixedRate(new RumbleTask(), 10, 20);
+
+        aButton = new DTTrigger(controller::getAButton);
+        bButton = new DTTrigger(controller::getBButton);
+        xButton = new DTTrigger(controller::getXButton);
+        yButton = new DTTrigger(controller::getYButton);
+        leftMenuButton = new DTTrigger(controller::getBackButton);
+        rightMenuButton = new DTTrigger(controller::getStartButton);
+        leftBumper = new DTTrigger(controller::getLeftBumper);
+        rightBumper = new DTTrigger(controller::getRightBumper);
+        leftStickButton = new DTTrigger(controller::getLeftStickButton);
+        rightStickButton = new DTTrigger(controller::getRightStickButton);
+
+        dpadNone = new DTTrigger(() -> controller.getPOV() == -1);
+        dpadUp = new DTTrigger(() -> controller.getPOV() == 0);
+        dpadUpRight = new DTTrigger(() -> controller.getPOV() == 45);
+        dpadRight = new DTTrigger(() -> controller.getPOV() == 90);
+        dpadDownRight = new DTTrigger(() -> controller.getPOV() == 135);
+        dpadDown = new DTTrigger(() -> controller.getPOV() == 180);
+        dpadDownLeft = new DTTrigger(() -> controller.getPOV() == 225);
+        dpadLeft = new DTTrigger(() -> controller.getPOV() == 270);
+        dpadUpLeft = new DTTrigger(() -> controller.getPOV() == 315);
+
+        leftStickX = new DTAxis(controller::getLeftX);
+        leftStickY = new DTAxis(controller::getLeftY);
+        rightStickX = new DTAxis(controller::getRightX);
+        rightStickY = new DTAxis(controller::getRightY);
+        leftTrigger = new DTAxis(controller::getLeftTriggerAxis);
+        rightTrigger = new DTAxis(controller::getRightTriggerAxis);
```

Comment:
Could dpadUp/Down and dpadLeft/Right be considered axes that only take on values of -1, 0, and 1?

Reply:
Potentially... this is a good addition.

Reply:
I’m also going to specify that dpad UpLeft counts as both up and left in this case (so x=-1, y=1), and similarly for the other corners. This can be reverted if needed. Another idea might be to make them +/- sqrt(2)/2, but that might just be unnecessarily complex (although technically correct).